app/presenters/order_detail_notice_presenter.rb

```diff
@@ -0,0 +1,85 @@
+class OrderDetailNoticePresenter < DelegateClass(OrderDetail)
+
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::OutputSafetyHelper
+
+  def statuses
+    [].tap do |statuses|
+      statuses << Notice.new(:in_review) if in_review?
+      statuses << Notice.new(:in_dispute) if in_dispute?
+      statuses << Notice.new(:can_reconcile) if can_reconcile_journaled?
+      statuses << Notice.new(:in_open_journal) if in_open_journal?
+    end
+  end
+
```

Comment:
I kind of hate `tap`, to be honest. (In general, not just here--which is a good case to use it if  ever there was one.)  It's just more cognitive effort to figure out exactly what is happening/what is being returned in this method with `tap` vs the local variable version, which makes it super obvious what the return value on the method is.

`tap` isn't a deal-breaker for me though.